BTCPayServer/wwwroot/js/store-reports.js

```diff
@@ -183,9 +183,9 @@ function downloadCSV() {
     if (!origData) return;
     const data = clone(origData);
 
-    // Convert ISO8601 dates to YYYY-MM-DD HH:mm:ss so the CSV easily integrate with Excel
+    // Convert ISO8601 dates to "dddd, MMMM Do YYYY, h:mm:ss a" to display full date plus time in 12 hour format so the CSV easily integrate with Excel
     modifyFields(srv.result.fields, data, 'amount', displayValue)
-    modifyFields(srv.result.fields, data, 'datetime', v => v? moment(v).format('YYYY-MM-DD hh:mm:ss'): v);
+    modifyFields(srv.result.fields, data, 'datetime', v => v ? moment(v).format("dddd, MMMM Do YYYY, h:mm:ss a"): v);
```

Comment:
To not break the curent format I'd suggest to simply use proper ISO format with added timezone info, like this: 

```suggestion
    modifyFields(srv.result.fields, data, 'datetime', v => v ? (new Date(v)).toISOString().replace('T', ' '));
```

Not sure if `v` is a `Date` already, if so it could be `v.toISOString()`.

Reply:
we shouldn't do string manipulations to format dates when we have a library to do things properly.